course_catalog/etl/see.py

```diff
@@ -0,0 +1,274 @@
+
+
+def _parse_topics(course):
+    """
+    Extract topics for a course
+
+    Args:
+        course(Tag): The BeautifulSoup tag containing course data
+
+    Returns:
+          list of str: List of topic names
+
+    """
+    topics = []
+    for classname in ("primary-topics", "secondary-topics"):
+        try:
+            topics.extend(
+                [
+                    {"name": strip_extra_whitespace(topic)}
+                    for topic in course.find("span", {"class": classname})
+                    .parent.find(text=True, recursive=False)
+                    .split(",")
+                ]
+            )
+        except AttributeError:
+            pass
+    return topics
+
+
+def _parse_price(course):
+    """
+    Extract the course price as a list
+
+    Args:
+        course(Tag): The BeautifulSoup tag containing course data
+
+    Returns:
+          list of int: List containing 1 integer (or empty)
+    """
+    prices = []
+    price_text = course.find("strong", text=re.compile(r"Tuition:.*")).parent.find(
+        text=True, recursive=False
+    )
+    if price_text:
+        price_match = re.search(r"([\d,]+)", price_text)
+        if price_match:
+            return [{"price": Decimal(price_match.group(0).replace(",", ""))}]
+    return prices
+
+
+def _parse_run_dates(course):
+    """
+    Extracts the start and end dates for each course run
+
+    Args:
+        course(Tag): The BeautifulSoup tag containing course data
+
+    Returns:
+        list of tuple: List of start and end date tuples for each course run
+
+    """
+    # Start and end dates in same month (Jun 18-19, 2020)
+    pattern_1_month = re.compile(
+        r"(?P<start_m>\w+)\s+(?P<start_d>\d+)-?(?P<end_d>\d+)?,\s*(?P<year>\d{4})$"
+    )
+    # Start and end dates in different months, same year (Jun 18 - Jul 18, 2020)
+    pattern_1_year = re.compile(
+        r"(?P<start_m>\w+)\s+(?P<start_d>\d+)\s*\-\s*(?P<end_m>\w+)\s+(?P<end_d>\d+),\s*(?P<year>\d{4})$"
+    )
+    # Start and end dates in different years (Dec 21, 2020-Jan 10,2021)
+    pattern_2_years = re.compile(
+        r"(?P<start_m>\w+)\s+(?P<start_d>\d+),\s*(?P<start_y>\d{4})\s*-\s*(?P<end_m>\w+)\s+(?P<end_d>\d+),\s*(?P<end_y>\d{4})$"
+    )
+    date_strings = [
+        rundate.strip() for rundate in course.findAll("li")[3].get_text().split("|")
+    ]
+    runs = []
+    for date_string in date_strings:
+        match = re.match(pattern_1_month, date_string)
+        if match:
+            start_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('start_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            end_date = datetime.strptime(
+                f"{match.group('start_m')} {match.group('end_d')} {match.group('year')}",
+                "%b %d %Y",
+            )
+            runs.append(
+                (start_date.replace(tzinfo=pytz.utc), end_date.replace(tzinfo=pytz.utc))
```

Comment:
Should we use `(start_date.date(), end_date.date())` instead? If we don't need the exact time then we don't need to deal with the time zone.

Reply:
It works but I get a bunch of warnings in the console:

/usr/local/lib/python3.7/site-packages/django/db/models/fields/__init__.py:1369: RuntimeWarning: DateTimeField LearningResourceRun.best_start_date received a naive datetime (2020-07-29 00:00:00) while time zone support is active.
  RuntimeWarning)

Reply:
Yeah, I guess if the field is expecting a datetime then we need to set a timezone explicitly.